### DISCUSSION TITLE
useViewportSize staying at 0 on intitial render

### DISCUSSION THREAD

**Discussion by BMCwebdev** [adobe/react-spectrum#6661] [Q&A]
Hey there all. I was trying to use useViewportSize but when the page would render, the width value would stay at 0. I swapped out for a basic version, but would like to swap back to useViewportSize as it looks to be very helpful especially when we get to iOS issues.

The issue presents once the component was put to use in a Next.js app. But it looks like useViewportSize is set up for SSR detection.

Here are the relevant bits of code. Anyone see anything needs fixin'?

```
  const { width } = useViewportSize();
  const [isNavOpen, setIsNavOpen] = useState(() => width > maxLargeValue);

  // Update isNavOpen state when width or focusOnOpen changes
  useEffect(() => {
    if (width > maxLargeValue) {
      setIsNavOpen(true);
      setFocusOnOpen(false);
    } else if (width <= maxLargeValue && !focusOnOpen) {
      setIsNavOpen(false);
    }
  }, [width, focusOnOpen]);
```

**Comment by snowystinger** [comment]
Thanks for the issue, would you mind providing a stackblitz or github repo which demonstrates the issue? I'm not seeing the issue you're describing.
